sparse_strips/vello_sparse_tests/tests/util.rs

```diff
@@ -286,6 +293,102 @@ pub(crate) fn check_ref(
     }
 }
 
+#[cfg(target_arch = "wasm32")]
+pub(crate) fn check_ref(
+    ctx: &impl Renderer,
+    // The name of the test.
+    test_name: &str,
+    _: &str,
+    // Tolerance for pixel differences.
+    threshold: u8,
+    // Must be `false` on `wasm32` as reference image cannot be written to filesystem.
+    is_reference: bool,
+    render_mode: RenderMode,
+    ref_data: &[u8],
+) {
+    assert!(!is_reference, "WASM cannot create new reference images");
+
+    let pixmap = render_pixmap(ctx, render_mode);
+    let encoded_image = pixmap_to_png(pixmap, ctx.width() as u32, ctx.height() as u32);
+    let actual = load_from_memory(&encoded_image).unwrap().into_rgba8();
+
+    let ref_image = load_from_memory(ref_data).unwrap().into_rgba8();
+
+    let diff_image = get_diff(&ref_image, &actual, threshold);
+    if let Some(ref img) = diff_image {
+        append_diff_image_to_browser_document(test_name, img);
+        panic!("test didn't match reference image. Scroll to bottom of browser to view diff.");
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+fn append_diff_image_to_browser_document(test_name: &str, diff_image: &RgbaImage) {
```

Comment:
I think that this test name should be using the `specific_test_name` from elsewhere, so that you can tell the difference between CPU only, hybrid and Webgl running